bluesky/plans.py

```diff
@@ -60,15 +60,30 @@ def plan_mutator(plan, msg_proc):
     plan : generator
         a generator that yields messages (`Msg` objects)
     msg_proc : callable
-        Functions that takes in a message and returns replacement messages.
-        Signatures:
-
-        * msg -> None, None (no op)
-        * msg -> gen, None (mutate and/or insert before current message;
-          last message in gen must invoke a response compatible
-          with original msg)
-        * msg -> gen, tail (same as above, but insert some messages *after*)
-        * msg -> None, tail (illegal -- raises RuntimeError)
+        This function takes in a message and specifies messages(s) to replace
+        it with. The function must account for what type of response the
+        message would prompt. For example, an 'open_run' message causes the
+        RunEngine to send a uid string back to the plan, while a 'set' message
```

Comment:
Maybe 'yields a uid string back to the plan'. Correct me if I misunderstood this part. I feel that this is modifying the generator behaviour, and thus the RunEngine does not need to be introduced.